tests/test_image.h

```diff
@@ -0,0 +1,283 @@
+
+	// Load using load()
+	Ref<Image> image_load = memnew(Image());
+	err = image_load->load(save_path_png);
+	CHECK_MESSAGE(
+			err == OK,
+			"The image should load successfully using load().");
+	CHECK_MESSAGE(
+			image->get_data() == image_load->get_data(),
+			"The loaded image should have the same data as the one that got saved.");
+
+	const String base_dir = OS::get_singleton()->get_executable_path().get_base_dir();
+
+	// Load BMP
+	const String input_bmp_path = base_dir.plus_file("../tests/test_image/icon.bmp");
+	Ref<Image> image_bmp = memnew(Image());
+	FileAccessRef f_bmp = FileAccess::open(input_bmp_path, FileAccess::READ, &err);
+	PackedByteArray data_bmp;
+	data_bmp.resize(f_bmp->get_len() + 1);
+	f_bmp->get_buffer(data_bmp.ptrw(), f_bmp->get_len());
+	CHECK_MESSAGE(
+			image_bmp->load_bmp_from_buffer(data_bmp) == OK,
+			"The BMP image should load successfully.");
+
+	// Load JPG
+	const String input_jpg_path = base_dir.plus_file("../tests/test_image/icon.jpg");
+	Ref<Image> image_jpg = memnew(Image());
+	FileAccessRef f_jpg = FileAccess::open(input_jpg_path, FileAccess::READ, &err);
+	PackedByteArray data_jpg;
+	data_jpg.resize(f_jpg->get_len() + 1);
+	f_jpg->get_buffer(data_jpg.ptrw(), f_jpg->get_len());
+	CHECK_MESSAGE(
+			image_jpg->load_jpg_from_buffer(data_jpg) == OK,
+			"The JPG image should load successfully.");
+
+	// Load WEBP
+	const String input_webp_path = base_dir.plus_file("../tests/test_image/icon.webp");
+	Ref<Image> image_webp = memnew(Image());
+	FileAccessRef f_webp = FileAccess::open(input_webp_path, FileAccess::READ, &err);
+	PackedByteArray data_webp;
+	data_webp.resize(f_webp->get_len() + 1);
+	f_webp->get_buffer(data_webp.ptrw(), f_webp->get_len());
+	CHECK_MESSAGE(
+			image_webp->load_webp_from_buffer(data_webp) == OK,
+			"The WEBP image should load successfully.");
+
+	// Load PNG
+	const String input_png_path = base_dir.plus_file("../tests/test_image/icon.png");
+	Ref<Image> image_png = memnew(Image());
+	FileAccessRef f_png = FileAccess::open(input_png_path, FileAccess::READ, &err);
+	PackedByteArray data_png;
+	data_png.resize(f_png->get_len() + 1);
+	f_png->get_buffer(data_png.ptrw(), f_png->get_len());
+	CHECK_MESSAGE(
+			image_png->load_png_from_buffer(data_png) == OK,
+			"The PNG image should load successfully.");
+
+	// Load TGA
+	const String input_tga_path = base_dir.plus_file("../tests/test_image/icon.tga");
+	Ref<Image> image_tga = memnew(Image());
+	FileAccessRef f_tga = FileAccess::open(input_tga_path, FileAccess::READ, &err);
+	PackedByteArray data_tga;
+	data_tga.resize(f_tga->get_len() + 1);
+	f_tga->get_buffer(data_tga.ptrw(), f_tga->get_len());
+	CHECK_MESSAGE(
+			image_tga->load_tga_from_buffer(data_tga) == OK,
+			"The TGA image should load successfully.");
+}
+
+TEST_CASE("[Image] Basic getters") {
+	Ref<Image> image = memnew(Image(8, 4, false, Image::FORMAT_LA8));
+	CHECK_MESSAGE(
+			image->get_width() == 8,
+			"get_width() should return the expected value.");
```

Comment:
These type of messages are redundant in my opinion ("X expected to be Y"). I'd just use `CHECK()` macro for those. My rule of thumb is: if I start to write rationale which contains the "expected" word, I just replace `CHECK_MESSAGE` to `CHECK`.

Reply:
All right, I changed some of the checks to just `CHECK()`.